src/modules/did/did.processor.ts

```diff
@@ -23,16 +39,30 @@ export class DIDProcessor {
   @Process(ADD_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentAddition(job: Job<string>) {
     this.logger.debug(`processing cache add for ${job.data}`);
-    await this.didService.addCachedDocument(job.data);
+    const doc = await this.didService.addCachedDocument(job.data);
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
   }
 
   @Process(UPDATE_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentRefresh(job: Job<string>) {
     this.logger.debug(`processing cache refresh for ${job.data}`);
-    if (this.configService.get('DID_SYNC_MODE_FULL') === 'true') {
-      await this.didService.addCachedDocument(job.data, true);
+    let doc: DIDDocumentEntity;
+    if (this.config.get('DID_SYNC_MODE_FULL') === 'true') {
+      doc = await this.didService.addCachedDocument(job.data, true);
     } else {
-      await this.didService.incrementalRefreshCachedDocument(job.data);
+      doc = await this.didService.incrementalRefreshCachedDocument(job.data);
+    }
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
+  }
+
+  private async pinClaims(cids: string[]) {
+    for await (const cid of cids) {
```

Comment:
I think there is important difference here. Unlike regular `for` the `await` variance awaits result of each iteration. This is stated in the `Description` section

> If the iterable is a sync iterable that yields promises, for await...of would produce a sequence of resolved values, while for...of would produce a sequence of promises

This is important for us, because we expect all claims are persisted when job complets

Reply:
Hmm, I think I'm still missing your point. The quote you mentioned is about the `iterable`, referring to this code snippet from the documentation https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of#syntax
```
for await (variable of iterable)
  statement
```

And the `iterable` in our case is `cids` which is of type `string[]`, so the `iterable` is not a "sync iterable that yields promises", right?

Reply:
Yes, you are right. I confused promise yieled by iterator with loop body